IQSS/dataverse-frontend
# 264 - Add the AccessFileMenu to File Page

## What this PR does / why we need it:
This PR adds the AccessFileMenu to the File Page. Since the menu component already existed and was being used in the Files Table from the Dataset Page in its button icon form, I refactored the necessary components to rehuse the component in the File Page.

To refactor the AccessFileMenu component I first needed to extract the File Metadata from the File model. That's why this branch parent is [refactor/263-extract-file-metadata](https://github.com/IQSS/dataverse-frontend/tree/refactor/263-extract-file-metadata)

## Which issue(s) this PR closes:

- Closes #264 

## Special notes for your reviewer:
Take into account that the button already existed in its icon button shape in the Dataset Page:
![Captura de pantalla 2024-01-24 a las 16 48 52](https://github.com/IQSS/dataverse-frontend/assets/23359572/5fde3b81-b14b-4c7d-b5fe-027ffdb13435)

## Suggestions on how to test this:
1. Install the necessary dependencies using `npm install`.
5. Build the design system with `cd packages/design-system && npm run build`.
6. Navigate back to the root directory with `cd ../../`.
7. Start Storybook with `npm run storybook`.
8. Visit [the File Page in the Storybook](http://localhost:6006/?path=/story/pages-file--default) and check that the Access File Menu looks as expected
9. For detailed menu stories visit [AccessFileMenu stories](http://localhost:6006/?path=/story/sections-file-page-action-buttons-accessfilemenu--default)
10. To ensure that the refactor didn't affect the button in the Files Table, which is the same button but in its button icon shape, take a look at the [Access File Menu in the Dataset Page story](http://localhost:6006/?path=/story/pages-dataset--default)

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
![Captura de pantalla 2023-12-14 a las 11 41 12](https://github.com/IQSS/dataverse-frontend/assets/23359572/917df704-e642-424f-986c-559c77cc82bc)

## Is there a release notes update needed for this change?:
No

## Additional documentation:
